src/libra_py/pdos.py

```diff
@@ -237,4 +236,4 @@ def QE_pdos(prefix, emin, emax, de, projections, Ef, outfile_prefix, do_convolve
     if nspin == 2:
         return E, pDOSa, pDOSb
     else:
-        return E, pDOSa, None
+        return E, pDOSa, pDOSa

```

Comment:
this will return 2 identical references, so if one changes the returned 3-rd values, it would affect the return 2-nd value. You need to return **copies** of the object, not the object itself (this works as a reference)